File: library/src/main/java/tw0reck1/sonar/Sonar.java
package tw0reck1.sonar;

import java.util.Collection;

public interface Sonar {

    void setPoints(Collection<SonarPoint> points);

    void addPoints(SonarPoint...points);

    void setColor(int color);

    void setStrokeWidth(float strokeWidth);

    void setThinStrokeWidth(float thinStrokeWidth);

    void setFontSize(float fontSize);

    void setThinFontSize(float thinFontSize);

    void setPointSize(float pointSize);

    void setSizes(float strokeWidth, float thinStrokeWidth, float fontSize, float thinFontSize, float pointSize);

    void startAnimation();

    void stopAnimation();

}

File: library/src/main/java/tw0reck1/sonar/Sonar.kt
package tw0reck1.sonar

interface Sonar {
    fun setPoints(points: Iterable<SonarPoint>)
    fun addPoints(vararg points: SonarPoint)
    fun setColor(color: Int)
    fun setStrokeWidth(strokeWidth: Float)
    fun setThinStrokeWidth(thinStrokeWidth: Float)
    fun setFontSize(fontSize: Float)
    fun setThinFontSize(thinFontSize: Float)
    fun setPointSize(pointSize: Float)
    fun setSizes(
        strokeWidth: Float,
        thinStrokeWidth: Float,
        fontSize: Float,
        thinFontSize: Float,
        pointSize: Float
    )
    fun startAnimation()
    fun stopAnimation()
}
